src/Functions/if.cpp

```diff
@@ -25,6 +25,8 @@
 #include <Functions/FunctionFactory.h>
 #include <type_traits>
 
+#pragma clang diagnostic ignored "-Wundefined-reinterpret-cast"
```

Comment:
Let's remove this:

```patch
diff --git a/src/Functions/if.cpp b/src/Functions/if.cpp
index fba50385923..f8f44baead7 100644
--- a/src/Functions/if.cpp
+++ b/src/Functions/if.cpp
@@ -25,7 +25,7 @@
 #include <Functions/FunctionFactory.h>
 #include <type_traits>
 
-#pragma clang diagnostic ignored "-Wundefined-reinterpret-cast"
+//#pragma clang diagnostic ignored "-Wundefined-reinterpret-cast"
 
 namespace DB
 {
@@ -59,7 +59,11 @@ concept is_native_int_or_decimal_v
     auto mask = static_cast<UIntType>(static_cast<IntType>(vc) - 1); \
     auto new_a = static_cast<ResultType>(va); \
     auto new_b = static_cast<ResultType>(vb); \
-    auto tmp = (~mask & (*reinterpret_cast<UIntType *>(&new_a))) | (mask & (*reinterpret_cast<UIntType *>(&new_b))); \
+    UIntType uint_a; \
+    std::memcpy(&uint_a, &new_a, sizeof(UIntType)); \
+    UIntType uint_b; \
+    std::memcpy(&uint_b, &new_b, sizeof(UIntType)); \
+    UIntType tmp = (~mask & uint_a) | (mask & uint_b); \
     (vr) = *(reinterpret_cast<ResultType *>(&tmp));
 
 template <typename ArrayCond, typename ArrayA, typename ArrayB, typename ArrayResult, typename ResultType>
```

Should have the same performance in x86_64 and not introduce undefined behaviour.